home/utils.py

```diff
@@ -631,3 +432,284 @@ def export_csv_content(queryset: PageQuerySet, response: HttpResponse) -> None:
     writer = csv.writer(response)
     for row in content_sheet:
         writer.writerow(row)
+
+
+@dataclass
+class ContentSheetRow:
+    side_panel_column_1: str = ""
+    side_panel_column_2: str = ""
+    side_panel_column_3: str = ""
+    side_panel_column_4: str = ""
+    side_panel_column_5: str = ""
+    side_panel_message_number: int = 0
+    parent: str = ""
+    web_title: str = ""
+    web_subtitle: str = ""
+    web_body: str = ""
+    whatsapp_title: str = ""
+    whatsapp_body: str = ""
+    messenger_title: str = ""
+    messenger_body: str = ""
+    viber_title: str = ""
+    viber_body: str = ""
+    translation_tag: str = ""
+    tags: str = ""
+    quick_replies: str = ""
+    triggers: str = ""
+    locale: str = ""
+    next_prompt: str = ""
+    image_link: str = ""
+    doc_link: str = ""
+    media_link: str = ""
+    related_pages: str = ""
+    tuple_of_extra_rows: tuple = ()
+
+    def __init__(
+        self,
+        page: ContentPage,
+        structure_string: str,
+        side_panel_message_number: int = 0,
+    ) -> None:
```

Comment:
I think with dataclasses, the convention is to let the dataclass create the `__init__` function. And if you need to create an instance of it some other way, then you use `classmethod`s, eg.
```
@dataclass
class ContentSheetRow:
    side_panel_column_1: str = ""
    ...
    
    @classmethod
    def from_page(cls, page: ContentPage, structure_string: str, ...) -> "ContentSheetRow":
        return cls(
            side_panel_column_1=...,
            ...
        )
```

Reply:
ah thanks, I will look into it